File: frontend/src/utils/accountRoles.js
export const CUSTOMER_ROLE = 'customer'
export const ADMIN_ROLE = 'admin'
export const SUPER_ADMIN_ROLE = 'super_admin'

export const normalizeAccountRole = (value) => {
  const role = String(value || '').trim().toLowerCase().replaceAll('-', '_')
  return role || CUSTOMER_ROLE
}

export const canUseCustomerApp = (role) =>
  [CUSTOMER_ROLE, ADMIN_ROLE, SUPER_ADMIN_ROLE].includes(normalizeAccountRole(role))

export const canUseAdminTools = (account = {}) => {
  const role = normalizeAccountRole(account.role)
  const adminRole = normalizeAccountRole(account.admin_role)
  return role === ADMIN_ROLE || role === SUPER_ADMIN_ROLE || adminRole === SUPER_ADMIN_ROLE
}

File: foodnova-customer-app/lib/shared/auth/account_roles.dart
const customerRole = 'customer';
const adminRole = 'admin';
const superAdminRole = 'super_admin';

const shoppingAppRoles = {
  customerRole,
  adminRole,
  superAdminRole,
};

const adminToolRoles = {
  adminRole,
  superAdminRole,
};

String normalizeAccountRole(Object? value) {
  if (value == null) return customerRole;
  final role = '$value'.trim().toLowerCase().replaceAll('-', '_');
  return role.isEmpty ? customerRole : role;
}

bool canUseCustomerApp(Object? role) {
  return shoppingAppRoles.contains(normalizeAccountRole(role));
}

bool canUseAdminTools(Object? role) {
  return adminToolRoles.contains(normalizeAccountRole(role));
}
